crates/bls12381/src/curves/g1.rs

```diff
@@ -127,6 +127,34 @@ impl<F: PrimeField + PrimeFieldBits> G1Point<F> {
         Ok(Self { x: xr, y: yr })
     }
 
+    //Implements constraint: (y_a + y_c) * (x_b - x_a) - (y_b - y_a)*(x_a - x_c) = 0 mod p
+    //used to show (x_a, y_a), (x_b, y_b), (x_c, -y_c) are co-linear
+    pub fn collinearcheck<CS>(&self, cs: &mut CS, b: &Self, c: &Self) -> Result<(), SynthesisError>
+    where
+        CS: ConstraintSystem<F>,
+    {
+        let (a, b, c) = (self, b, c);
+        let cs = &mut cs.namespace(|| "G1::collinearcheck(a, b, c)");
+
+        // compute leftside = (y_a + y_c) * (x_b - x_a)
+        let aycy = a.y.add(&mut cs.namespace(|| "aycy <- a.y + c.y"), &c.y)?;
+        let bxax = b.x.sub(&mut cs.namespace(|| "bxax <- b.x - a.x"), &a.x)?;
+        let leftside = aycy.mul(&mut cs.namespace(|| "leftside <- aycy * bxax"), &bxax)?;
+
+        //compute rightside = (y_b - y_a)*(x_a - x_c)
+        let byay = b.y.sub(&mut cs.namespace(|| "byay <- b.y - a.y"), &a.y)?;
+        let axcx = a.x.sub(&mut cs.namespace(|| "axcx <- a.x - c.x"), &c.x)?;
+        let rightside = byay.mul(&mut cs.namespace(|| "rightside <- byay * axcx"), &axcx)?;
+
+        FpElement::assert_is_equal(
+            &mut cs.namespace(|| "leftside =? rightside"),
+            &leftside,
+            &rightside,
+        );
```

Comment:
This returns a Result, so this is missing a `?` here
```suggestion
        )?;
```